plugin/src/main/java/io/jenkins/plugins/coverage/model/CoverageNode.java

```diff
@@ -319,6 +358,22 @@ public Coverage getCoverage(final CoverageMetric searchMetric) {
         return childNodes;
     }
 
+    /**
+     * Returns recursively all nodes of the instance {@link FileCoverageNode}.
+     *
+     * @return all file coverage nodes
+     */
+    public List<FileCoverageNode> getAllFileCoverageNodes() {
```

Comment:
Is this method required? Wouldn't it be simpler to use `getAll` and work on the `CoverageNode` types?

Reply:
I wrote this method since otherwise, there has to be a type check everytime it is used. Otherwise, there will be cast exception when for example the reference build has been built with an older release, where `FileCoverageNode` did not exist and `getAll` returns nodes of the type `CoverageNode` only.

Reply:
But why do you need to cast anyway? You are using only methods of `CoverageNode`.

Reply:
I use this method in classes like `FileCoverageDeltaProcessor` where I got the root of the tree as `CoverageNode` and I need all `FileCoverageNode` of the tree since only this subclass stores the information about changed lines or the coverage per line - here, I would have to cast to access these information.
For example, when the reference root has been built with an older version, this cast would fail if I do not check it first. Since there are many cases like this, I wrote this method `getAllFileCoverageNodes`.